### ISSUE TITLE
Problem:  `mina-indexer::receiver detects_new_block_copied` is flakey when nextest executes tests concurrently

### ISSUE THREAD

**Issue by trevorbernard** [Granola-Team/mina-indexer#287]
The test is flakey when the test runner is run in parallel

**Comment by robinbb** [comment]
... and sometimes when it's run with only 1 thread.

**Comment by robinbb** [comment]
... this is starting to cause a few CI failures.

**Comment by trevorbernard** [comment]
I suggest removing these unit tests and replace them with a more fitting regression test. But unfortunately the regression test case will fail currently as we have a bug in the system. Until that is resolved these shall remain `flakey`.

**Comment by robinbb** [comment]
Unit tests are nice to have.

**Comment by jenr24G** [comment]
the unit test here ends up just testing the functionality of the library we're using under the hood. The `BlockReceiver` does nothing more than spawning a filesystem worker from https://docs.rs/watchexec/latest/watchexec/fs/fn.worker.html I'm starting to think this bug is a bug in the underlying library

**Comment by robinbb** [comment]
This is unlikely, but possible. In any case, the bug in our code remains.

**Comment by Isaac-DeFrain** [comment]
I believe https://github.com/Granola-Team/mina-indexer/pull/343 addresses the flakiness (tests are consistently passing on my machine). Someone else should definitely corroborate on their machine.

**Comment by trevorbernard** [comment]
Issue looks to be resolved
